Shoko.Server/API/v3/Models/Common/TotalList.cs

```diff
@@ -0,0 +1,45 @@
+using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.ComponentModel.DataAnnotations;
+
+#nullable enable
+namespace Shoko.Server.API.v3.Models.Common
+{
+    /// <summary>
+    /// A list with a total count and a sliced page of the total or the total
+    /// list.
+    /// </summary>
+    public class TotalList<T>
```

Comment:
It's a nitpick, but I would name it something like ListResult, to match more with the ActionResult naming scheme.